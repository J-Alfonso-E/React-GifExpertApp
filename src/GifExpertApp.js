import React, { useState } from "react"
import { AgregarCategoria } from "./Componentes/AgregarCategoria";
import { GifGrid } from "./Componentes/GifGrid";


const GifExpertApp = () => {

    const [categorias, setCategorias] = useState(['Halo Infinite']);
    

    /*const AgregarCategorias = () => {
        setCatergorias([...categorias, '9 Years of Shadows']); // Operador Spread  https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Operators/Spread_syntax

        //Segunda forma de Regresar el valor

        //setCatergorias(cats => [...cats, '9 Years of Shadows']);
    }*/
    return (
        <>
            <h2>Gif Expert App</h2>
            <AgregarCategoria  EnviarCategoria={setCategorias}/> {/* Envio de una funcion al componente */}
            <hr />

            

            <ol>
                {
                categorias.map(Catg => (
                    <GifGrid
                    key = {Catg}
                    categoria = {Catg}
                    />
                ))
            }
            </ol>
        </>
    );
}


export default GifExpertApp;