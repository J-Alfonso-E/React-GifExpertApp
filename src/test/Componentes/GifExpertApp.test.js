import {shallow} from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('Pruebas de GifExpertApp', () => {
    test('SnapShot pagina Principal', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Listas de Categorias', () => {
        const categorias = ['Tales Arise', 'Gears 5'];

        const wrapper = shallow(<GifExpertApp CatPredeterminadas={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    });


});