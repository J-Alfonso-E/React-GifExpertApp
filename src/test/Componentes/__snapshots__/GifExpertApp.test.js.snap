// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp Listas de Categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AgregarCategoria
    EnviarCategoria={[Function]}
  />
   
  <hr />
  <ol>
    <GifGrid
      categoria="Tales Arise"
      key="Tales Arise"
    />
    <GifGrid
      categoria="Gears 5"
      key="Gears 5"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de GifExpertApp SnapShot pagina Principal 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AgregarCategoria
    EnviarCategoria={[Function]}
  />
   
  <hr />
  <ol />
</Fragment>
`;
